.github/workflows/test.yml

```diff
@@ -24,6 +24,22 @@ jobs:
           node-version: ${{ matrix.node }}
       - run: npm ci
       - run: npm test
+  typescript_matrix:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        typescript:
+          - 4.1
+          - 4.2
```

Comment:
Until which version of support do we want to arrive?

Reply:
Those are the current supported versions. `>=4.1, < 5` (as per `package.json`)

Reply:
> Those are the current supported versions. `>=4.1, < 5` (as per `package.json`)

Yep, I meant to say if we should consider give support to older versions of TypeScript. Tried to look for official supported versions by TypeScript team but I did not find any, do you know if there is a place with as NodeJS has for example?

Reply:
We can't support older versions of TS here or in `@octokit/webhook-definitions`. Since we use features only available in 4.2+

Reply:
> We can't support older versions of TS here or in `@octokit/webhook-definitions`. Since we use features only available in 4.2+

Yep, I was wondering if we would like to support older versions of TS and if it's worth it. Specially for the exposed types to the world. The internal part is fine to use the LTS.

---

Let's go for `TS v4.1` and `TS 4.2` and opening a discussion for giving support to older versions of TS if you think is worth having the discussion :)

Reply:
These are used in types exported publically.
See the `HandlerFunction` type, it is used by `Webhook#on` function and the type itself uses `EmitterWebhookEvent` which is incompatible with any version lower than `TS 4.1`

Reply:
for other readers: [this is the feature](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#template-literal-types) which requires `TS >= 4.1`